src/core/ddsc/src/dds_whc.c

```diff
@@ -40,47 +38,47 @@
 #define USE_EHH 0
 
 
-struct whc_node {
-  struct whc_node *next_seq; /* next in this interval */
-  struct whc_node *prev_seq; /* prev in this interval */
+struct ddsi_whc_node {
```

Comment:
You're right, I missed that layering violation and just renamed the definition of the struct, thanks for catching this! In [6db1c56](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1483/commits/6db1c56421ab9492d318f7753250c29211897c68) I've tried to fix this by adding a base type `ddsi_whc_node`. Because the struct cannot be empty (MSVC) I've moved the `seq` member to the base type, but maybe other members can be moved there as well?